olive/passes/onnx/conversion.py

```diff
@@ -117,7 +117,25 @@ def _run_for_config(
             # only handle dict for now since we cannot get the name of the input from a list/tuple
             if isinstance(dummy_inputs, dict):
                 dummy_input_keys = set(dummy_inputs.keys())
+
+                # handle dummy inputs for hf model with past, which has past_key_values
+                # match input names in `past_key_values.(hidden_layer_num).(key|value)` pattern
+                try:
+                    from transformers.modeling_utils import PreTrainedModel
+
+                    if issubclass(type(pytorch_model), PreTrainedModel):
+                        for name, input in dummy_inputs.items():
+                            if isinstance(input, list):
+                                dummy_input_keys.update(
```

Comment:
I don't think this line really does anything. Since `unused_keys = dummy_input_keys - set(input_names)`, it doesn't matter whether the new keys are added to `dummy_input_keys`. What matter is the `name` is not present in `dummy_input_keys`. 

i think a better logic could be something like: 

```python
key_value_names = set(
    [f"{name}.{idx}.key" for idx in range(len(input))] + [f"{name}.{idx}.value" for idx in range(len(input))]
)
if key_value_names.issubset(set(input_names)):
    dummy_input_keys.discard(name)
```
This makes it certain that we are dealing with key value list.

Reply:
Yes, you are right. Sorry I didn't consider this situation. Thanks for your comment, I updated this part.